### PR TITLE
[bitnami/minio] add security context to provisioning job

### LINKED ISSUE #11616: [bitnami/minio] Add podSecurityContext and containerSecurityContext to provisioning job
### Name and Version

bitnami/minio 11.7.16

### What is the problem this feature will solve?

The provisioning-job.yaml template has no options to add a podSecurityContext and containerSecurityContext. Hence it is not possible to add a security context to the job.

### What is the feature you are proposing to solve the problem?

Add the securityContext to the pod and container like it is already added to the deployment and statefuleset templates.

Could be added like this:

``` yaml

{{- if .Values.provisioning.podSecurityContext.enabled }}
securityContext: {{- omit .Values.provisioning.podSecurityContext "enabled" | toYaml | nindent 8 }}
{{- end }}

```

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by Johannes-**
Signed-off-by: Johannes Voscort <johannes.voscort@codecentric.de>

### Description of the change

Added a security context to the pod and container of the provisioning job. 

### Benefits

Allows to set a custom security context to the provisioning job.

### Possible drawbacks

Can't think of any.

### Applicable issues

  - fixes #11616 

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/minio/templates/provisioning-job.yaml [hunk 1]
```diff
@@ -37,11 +37,17 @@ spec:
...
+      securityContext: {{- omit .Values.provisioning.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
       serviceAccountName: {{ template "minio.serviceAccountName" . }}
       initContainers:
         - name: wait-for-available-minio
           image: {{ include "minio.image" . }}
           imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          {{- if .Values.provisioning.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.provisioning.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
```
Comment (migruiz4) [inline bitnami/minio/templates/provisioning-job.yaml:50]: You added containerSecurityContext to the initContainer (wait-for-available-minio), but not the job main container (minio).
Is this intentional?

File: bitnami/minio/Chart.yaml [hunk 2]
```diff
@@ -25,4 +25,4 @@ name: minio
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/minio
   - https://min.io
-version: 11.7.18
+version: 11.7.19
```
Comment (migruiz4) [inline bitnami/minio/Chart.yaml:28]: Since this PR adds a new feature, please bump a minor version (`11.8.0`)

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thank you very much for your contribution @Johannes- !

Can you please take a look at my comments?

File: bitnami/minio/templates/provisioning-job.yaml (hunk 1 above)
Comment (Johannes-) [reply bitnami/minio/templates/provisioning-job.yaml:50]: Thanks! No, that was not intentional. I added it to the job container as well.

**Review by migruiz4 — APPROVED** [review]
Thank you very much for your contribution! Looks good to me!

**Pull request merged**